akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/PersistentEntity.scala

```diff
@@ -24,17 +24,17 @@ import akka.persistence.typed.javadsl.PersistentBehavior
 abstract class PersistentEntity[Command, Event, State >: Null] private (
   val entityTypeKey: EntityTypeKey[Command],
   val entityId:      String,
-  persistenceId:     PersistenceId, supervisorStrategy: Optional[BackoffSupervisorStrategy])
-  extends PersistentBehavior[Command, Event, State](persistenceId, supervisorStrategy) {
+  persistenceId:     PersistenceId, onPersistFailure: Optional[BackoffSupervisorStrategy])
+  extends PersistentBehavior[Command, Event, State](persistenceId, onPersistFailure) {
 
   def this(entityTypeKey: EntityTypeKey[Command], entityId: String) = {
     this(entityTypeKey, entityId,
       persistenceId = entityTypeKey.persistenceIdFrom(entityId), Optional.empty[BackoffSupervisorStrategy])
   }
 
-  def this(entityTypeKey: EntityTypeKey[Command], entityId: String, backoffSupervisorStrategy: BackoffSupervisorStrategy) = {
+  def this(entityTypeKey: EntityTypeKey[Command], entityId: String, onPersistFailure: BackoffSupervisorStrategy) = {
```

Comment:
onJournalFailure is maybe a better name. It is also triggered for a replay failing.

Reply:
Not sure the user is aware of the concept of journal. The primary API is Effect.persist and even if we also use it for recovery I think it's easiest to relate to persist.